src/NosCore.GameObject/Providers/NRunProvider/Handlers/TeleporterHandler.cs

```diff
@@ -0,0 +1,67 @@
+﻿//  __  _  __    __   ___ __  ___ ___  
+// |  \| |/__\ /' _/ / _//__\| _ \ __| 
+// | | ' | \/ |`._`.| \_| \/ | v / _|  
+// |_|\__|\__/ |___/ \__/\__/|_|_\___| 
+// 
+// Copyright (C) 2019 - NosCore
+// 
+// NosCore is a free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using NosCore.GameObject.ComponentEntities.Interfaces;
+using NosCore.GameObject.Networking.ClientSession;
+using ChickenAPI.Packets.Enumerations;
+using ChickenAPI.Packets.ClientPackets.Npcs;
+using NosCore.Core;
+using NosCore.GameObject.ComponentEntities.Extensions;
+using NosCore.Core.I18N;
+using NosCore.Data.Enumerations.I18N;
+
+namespace NosCore.GameObject.Providers.NRunProvider.Handlers
+{
+    public class TeleporterHandler : IEventHandler<Tuple<IAliveEntity, NrunPacket>, Tuple<IAliveEntity, NrunPacket>>
+    {
+        public bool Condition(Tuple<IAliveEntity, NrunPacket> item) => item.Item2.Runner == (NrunRunnerType)16
+            && item.Item1.MapInstance.Npcs.Find(s => s.Dialog >= 439 && s.Dialog <= 441) != null;
```

Comment:
You still don’t need the find and in this case it doesn’t work as you Are expecting it to work just use item.Item1 is MapNpc mapnpc and mapnpc.Dialog